lang/java/reef-io/src/main/java/org/apache/reef/io/network/naming/NameLookupClient.java

```diff
@@ -122,7 +101,7 @@ public NameLookupClient(final String serverAddr,
                           final Cache<Identifier, InetSocketAddress> cache,
                           final LocalAddressProvider localAddressProvider) {
     this(serverAddr, serverPort, timeout, factory, retryCount, retryTimeout,
-        cache, localAddressProvider, new MessagingTransportFactory());
+        cache, localAddressProvider, new MessagingTransportFactory(localAddressProvider));
```

Comment:
I suspect the intention is to make `NameLookupClient` injectable, similar to `NameClient`. This way we could reduce the number of constructors indeed. However, right now we can't do that, because `NameLookupClient` has public constructors not marked for deprecation as of 0.12. The best we can do is mark them for deprecation in 0.13 and remove in 0.14, which I'd like to do as part of making `NameLookupClient` injectable rather than here.